### ISSUE TITLE
Indentation for included template

### ISSUE THREAD

**Issue by egdd** [Antaris/RazorEngine#392]
Dear all,

I am making good progress with this great tool! As a requirement for my text templates I really have to get an idea of how to provide indentation. I want to generate text (no HTML) where indentation is important based on the calling parent template.

I included some custom methods in my _MyCustomTemplate(T)_ as described in the article [Extending the template Syntax](https://antaris.github.io/RazorEngine/TemplateBasics.html#Extending-the-template-Syntax).

What I can create with the engine right now is this

```
ParentContentBeginning
Son
Daughter
ParentContentEnd
```

What I want to create is this:

```
ParentContentBeginning
    Son
    Daughter
ParentContentEnd
```

The models for that would be:

```
public class Parent
{
    public Parent()
    {
        Children = new List<Child>();
    }

    IList<Child> Children { get; set;}
}

public class Child
{
    public string Name { get; set;}
}
```

The instances would be

```
new Parent() {
Children.Add("Son")
Children.Add("Daughter")
}
```

The templates would be something like this:

Parent.txt:

```
ParentContentBeginning
@foreach(var child in Model.Children)
{
    @IncludeWithIndentation("Child.txt, Model)
}
ParentContentEnd
```

Child.txt

```
@Model.Name
```

Basically, I want to indent the children. They could have children themselves as well, so they should indent their children as well (kind of recursively). The parent should specify the indentation for the children and add his indentation to the children's indendation as well.

Idea: If the method _IncludeWithIndentation_ is four spaces/tabs from the line start, all the children's content should be indented four spaces/tabs.

Two things I tried already:

- I tried to use RawStringFactory - however, no luck. Just literals end up here.
- In _MyCustomTemplate(T)_ I could imagine using a method like this _public IEncodeString IncludeWithIndentation(string indentation, string, templateName, object model))_ where the first parameter contains the spaces or tabs.

Any advice on that one?

**Comment by matthid** [comment]
Indentation is a difficult topic as there is no "correct" way to do it. Using the Indentation of the `@` doesn't lead to a properly formatted html document:

```
<body>
@if (someCondition) {
    foreach(var person in Model.Children)
    {
        @Include(...)
    }
}
<body>
```

To implement the `IncludeWithIndentation` you would change the `TemplateWriter` class (or add another class `TemplateWriterWithIndentation`) and replace the given delegate with another one writing the additional indent [here](https://github.com/Antaris/RazorEngine/blob/master/src/source/RazorEngine.Core/Templating/TemplateWriter.cs#L26). Use is [here](https://github.com/Antaris/RazorEngine/blob/master/src/source/RazorEngine.Core/Templating/TemplateBase.cs#L143).

It would look like this:
```
writerDelegate = textWriter => writer(new IndentedTextWriter(textWriter));
```
